server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/filemove/FileMoveDetectionStep.java

```diff
@@ -182,64 +195,133 @@ public void visitFile(Component file) {
       // FIXME computation of sourceHash and lineHashes might be done multiple times for some files: here, in ComputeFileSourceData, in
       // SourceHashRepository
       Component component = reportFilesByKey.get(fileKey);
-      SourceLinesHashesComputer linesHashesComputer = new SourceLinesHashesComputer();
-      try (CloseableIterator<String> lineIterator = sourceLinesRepository.readLines(component)) {
-        while (lineIterator.hasNext()) {
-          String line = lineIterator.next();
-          linesHashesComputer.addLine(line);
-        }
-      }
-      builder.put(fileKey, new File(component.getReportAttributes().getPath(), linesHashesComputer.getLineHashes()));
+      File file = new LazyFileImpl(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in constructor declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3066&issues=AWIvtD1FdSOV8npVrlz0&open=AWIvtD1FdSOV8npVrlz0)

[ISSUE_KEY]: AWIvtD1FdSOV8npVrlz0